src/tools/fuzzing/fuzzing.cpp

```diff
@@ -224,13 +224,14 @@ void TranslateToFuzzReader::setupMemory() {
     auto segment = builder.makeDataSegment();
     segment->memory = memory->name;
     segment->offset = builder.makeConst(int32_t(0));
-    segment->setName(Name::fromInt(0), false);
-    wasm.dataSegments.push_back(std::move(segment));
+    segment->setName(Names::getValidDataSegmentName(wasm, Name::fromInt(0)),
+                     false);
     auto num = upTo(USABLE_MEMORY * 2);
     for (size_t i = 0; i < num; i++) {
       auto value = upTo(512);
-      wasm.dataSegments[0]->data.push_back(value >= 256 ? 0 : (value & 0xff));
```

Comment:
Was this meant to be `wasm.dataSegment[i]`? It looks it should've been that way if the semantics are the same with the modified code..

Reply:
Hmm, I think it is pushing `num` chunks of binary data to the first segment here. `wasm.dataSegments[i]` might not exist - all we know is that `segment` exists.

What do you mean by the second sentence?

Reply:
Sorry I was confused.. Nevermind.